hypothesis-python/tests/cover/test_lookup_py37.py

```diff
@@ -0,0 +1,200 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import annotations
+
+import collections
+import collections.abc
+import contextlib
+import sys
+
+import pytest
+
+from hypothesis import assume, given, infer
+
+# This test file is skippped on Python <= 3.6, where `list[Elem]` is always a
+# SyntaxError.  On Python 3.7 and 3.8, `from __future__ import annotations` means
+# that the syntax is supported; but the feature fails at runtime.  On Python
+# 3.9 and later, it should all work.
+#
+# For details, see https://www.python.org/dev/peps/pep-0585/
+
+if sys.version_info < (3, 9):
+    pytestmark = pytest.mark.xfail(
+        raises=Exception, strict=True, reason="Requires Python 3.9 (PEP 585) or later."
+    )
+
+
+class Elem:
+    pass
+
+
+class Value:
+    pass
+
+
+def check(t, ex):
+    assert isinstance(ex, t)
+    assert all(isinstance(e, Elem) for e in ex)
+    assume(ex)
+
+
+@given(x=infer)
+def test_resolving_standard_tuple1_as_generic(x: tuple[Elem]):
+    check(tuple, x)
+
+
+@given(x=infer)
+def test_resolving_standard_tuple2_as_generic(x: tuple[Elem, Elem]):
+    check(tuple, x)
+
+
+@given(x=infer)
+def test_resolving_standard_tuple_variadic_as_generic(x: tuple[Elem, ...]):
+    check(tuple, x)
+
+
+@given(x=infer)
+def test_resolving_standard_list_as_generic(x: list[Elem]):
+    check(list, x)
+
+
+@given(x=infer)
+def test_resolving_standard_dict_as_generic(x: dict[Elem, Value]):
+    check(dict, x)
+    assert all(isinstance(e, Value) for e in x.values())
+
+
+@given(x=infer)
+def test_resolving_standard_set_as_generic(x: set[Elem]):
+    check(set, x)
+
+
+@given(x=infer)
+def test_resolving_standard_frozenset_as_generic(x: frozenset[Elem]):
+    check(frozenset, x)
+
+
+@given(x=infer)
+def test_resolving_standard_defaultdict_as_generic(
+    x: collections.defaultdict[Elem, Value]
+):
+    check(collections.defaultdict, x)
+    assert all(isinstance(e, Value) for e in x.values())
+
+
+@given(x=infer)
+def test_resolving_standard_iterable_as_generic(x: collections.abc.Iterable[Elem]):
+    check(collections.abc.Iterable, x)
+
+
+@given(x=infer)
+def test_resolving_standard_iterator_as_generic(x: collections.abc.Iterator[Elem]):
```

Comment:
What about `Generator`?

```
» python    
Python 3.9.0 (default, Oct 23 2020, 11:42:21) 
[Clang 11.0.0 (clang-1100.0.33.16)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import collections
>>> collections.abc.Generator[int, int, int]
collections.abc.Generator[int, int, int]
```

Reply:
We don't currently support `typing.Generator` (whoops!), so that and many other types will go in a follow-up PR based on https://github.com/Zac-HD/hypothesis/compare/pep585-annotations...register-more-types

I thought it was better to get PEP 585 support done soon, and keep each PR small enough to review.